crates/meroctl/src/cli/context/proposals.rs

```diff
@@ -0,0 +1,319 @@
+        } else {
+            for approver in &self.data {
+                let _ = table.add_row(vec![format!("{}", approver)]);
+            }
+        }
+
+        println!("{table}");
+    }
+}
+
+impl Report for GetProposalsResponse {
+    fn report(&self) {
+        let mut table = Table::new();
+        table.load_preset(comfy_table::presets::UTF8_FULL);
+        table.apply_modifier(comfy_table::modifiers::UTF8_ROUND_CORNERS);
+
+        let _ = table.set_header(vec![
+            Cell::new("ID").fg(Color::Blue),
+            Cell::new("Author").fg(Color::Blue),
+            Cell::new("Actions").fg(Color::Blue),
+        ]);
+
+        if self.data.is_empty() {
+            let _ = table.add_row(vec!["No proposals found", "", ""]);
+        } else {
+            for proposal in &self.data {
+                let _ = table.add_row(vec![
+                    format!("{}", proposal.id),
+                    format!("{}", proposal.author_id),
+                    format!("{}", proposal.actions.len()),
+                ]);
+            }
+        }
+
+        println!("{table}");
+    }
+}
+
+impl ProposalsCommand {
+    pub async fn run(&self, environment: &Environment) -> EyreResult<()> {
+        let config = load_config(&environment.args.home, &environment.args.node_name).await?;
+        let multiaddr = fetch_multiaddr(&config)?;
+        let client = Client::new();
+
+        match &self.command {
+            ProposalsSubcommand::List {
+                context,
+                offset,
+                limit,
+            } => {
+                let context_id = resolve_alias(multiaddr, &config.identity, *context, None)
+                    .await?
+                    .value()
+                    .cloned()
+                    .ok_or_eyre("unable to resolve context")?;
+
+                let args = serde_json::json!({
+                    "offset": offset,
+                    "limit": limit
+                });
+
+                self.list_proposals(
+                    environment,
+                    multiaddr,
+                    &client,
+                    &config.identity,
+                    context_id,
+                    args,
+                )
+                .await
+            }
+            ProposalsSubcommand::View {
+                proposal_id,
+                context,
+            } => {
+                let context_id = resolve_alias(multiaddr, &config.identity, *context, None)
+                    .await?
+                    .value()
+                    .cloned()
+                    .ok_or_eyre("unable to resolve context")?;
+
+                if let Err(e) = self
+                    .get_proposal(
+                        environment,
+                        multiaddr,
+                        &client,
+                        &config.identity,
+                        context_id,
+                        proposal_id,
+                    )
+                    .await
```

Comment:
```suggestion
                self.get_proposal(
                    environment,
                    multiaddr,
                    &client,
                    &config.identity,
                    context_id,
                    proposal_id,
                )
                .await?;
```

as mentioned in https://github.com/calimero-network/core/pull/1285#discussion_r2133073531